be/src/storage/sstable/table.cpp

```diff
@@ -250,11 +250,12 @@ Status Table::MultiGet(const ReadOptions& options, const Slice* keys, ForwardIt
         return false;
     };
 
-    for (auto it = begin; it != end; ++it) {
+    size_t i = 0;
+    for (auto it = begin; it != end; ++it, ++i) {
         auto& k = keys[*it];
         if (current_block_itr_ptr != nullptr && current_block_itr_ptr->Valid()) {
             // keep searching current block
-            if (search_in_block(k, *it)) {
+            if (search_in_block(k, i)) {
```

Comment:
let caller decide where the value should be written to.

```
search_in_block(const Slice &k, std::string &value);
```
if the `k` is found, do `value.assign(...)`, `search_in_block` will not be any interested of the `values` variable.

so is the `current_block_itr_ptr`, can be a parameter to `search_in_block` so that the `search_in_block` is totally stateless without catch any of the execution context.

Reply:
done